types/block.go

```diff
@@ -259,6 +264,107 @@ func mustPush(rowTree *nmt.NamespacedMerkleTree, id namespace.ID, data []byte) {
 	}
 }
 
+// PutBlock posts and pins erasured block data to IPFS using the provided
+// ipld.NodeAdder. Note: the erasured data is currently recomputed
+func (b *Block) PutBlock(ctx context.Context, nodeAdder format.NodeAdder) error {
+	if nodeAdder == nil {
+		return errors.New("no ipfs node adder provided")
+	}
+
+	// recompute the shares
+	namespacedShares := b.Data.computeShares()
```

Comment:
This is the only place the Block is used. This should be fairly easy to rip out the method into another package.